sdk/identity/ci.yml

```diff
@@ -35,6 +35,7 @@ extends:
         Path: sdk/identity/platform-matrix.json
         Selection: sparse
         GenerateVMJobs: true
+    TestProxy: true
```

Comment:
This should get things working for playback test pipelines! We'll also need to add two environment variables to `sdk/identity/tests.yml` to make sure the test proxy isn't used in that environment: https://github.com/Azure/azure-sdk-for-python/blob/8358194befbeb5bc5ddda4f44f5a92f56564cd07/sdk/keyvault/tests.yml#L39-L40